pinot-core/src/main/java/org/apache/pinot/core/upsert/UpsertProcessorUtil.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.upsert;
+
+import java.util.Map;
+import org.apache.pinot.core.realtime.impl.ThreadSafeMutableRoaringBitmap;
+import org.apache.pinot.spi.data.readers.PrimaryKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public final class UpsertProcessorUtil {
+  private static final Logger LOGGER = LoggerFactory.getLogger(UpsertProcessorUtil.class);
+
+  private UpsertProcessorUtil() {
+  }
+
+  public static void handleUpsert(PrimaryKey primaryKey, long timestamp, String segmentName, int docId, int partitionId,
+      Map<PrimaryKey, RecordLocation> primaryKeyIndex, ThreadSafeMutableRoaringBitmap validDocIndex,
+      TableUpsertMetadataManager upsertMetadataTableManager) {
+    RecordLocation location = new RecordLocation(segmentName, docId, timestamp);
+    // check local primary key index first
```

Comment:
local means in-memory segment? local is a bit ambiguous.

Reply:
I think all segments are in-memory? Changed it to for mutable segment.